common/entity/CourseAttendanceLog.ts

```diff
@@ -0,0 +1,45 @@
+import {
+    Column,
+    CreateDateColumn,
+    Entity,
+    PrimaryGeneratedColumn,
+    ManyToOne,
+    UpdateDateColumn
+} from "typeorm";
+import {Student} from "./Student";
+import {Pupil} from "./Pupil";
+import {Course} from "./Course";
+import {Subcourse} from "./Subcourse";
+import {Lecture} from "./Lecture";
+import {JoinColumn, OneToOne} from "typeorm/index";
+
+@Entity()
+export class CourseAttendanceLog {
+    @PrimaryGeneratedColumn()
+    id: number;
+
+    @Column({
+        default: () => "NOW()"
+    })
+    createdAt: Date;
+
+    @Column({ type: "timestamp", nullable: true })
+    updatedAt: Date;
+
+    @Column({ nullable: true })
+    attendedTime: number;
+
+    @Column({ nullable: true })
+    ip: string;
+
+    @ManyToOne(type => Pupil, pupil => pupil.courseAttendanceLog, {onUpdate: 'CASCADE', onDelete: 'CASCADE'})
+    pupil: Pupil;
+
+    @ManyToOne(type => Subcourse, subcourse => subcourse.courseAttendanceLog, {onUpdate: 'CASCADE', onDelete: 'CASCADE'})
+    subcourse: Subcourse;
```

Comment:
Do we need the `subcourse` here if we already have a `lecture` reference? This is actually storing duplicate information isn’t it? I think we can always access the sub course via a lecture property. Or what is the reason for adding that information here?

Reply:
No, `subcourse` is no longer required. After I got more understanding of the structure, I focused everything on the `lecture` and the `subcourse` became obsolete. Thanks for the hint!